### ISSUE TITLE
Support Required Flags With Type Annotations (And Weird Behavior with add_argument)

### ISSUE THREAD

**Issue by Chris3606** [innot/argparseDecorator#5]
`argparse` allows you to do something like this:
```py
parser.add_argument('--flag', help='A flag', required=True)
```

This causes argparse to treat the flag as required; eg. an error is raised if someone doesn't use it.

I could be mistaken, but as far as I can tell, there is no way to do this with the `argparsedecorator` library except for using the `add_argument` decorator.

In many cases I actually prefer to do CLI options this way as it makes the command-lines more readable IMO (positional arguments are ambiguous).  I can certainly use `add_argument` to do this, but having to use `**kwargs` is not the cleanest, and it seems like something that should be very easy to add to the type annotation parsing.

Related, the documentation of add_argument states:
> The decorated function must have an argument of the same name or use *args and **kwargs arguments to retrieve the value of these arguments.

I might be misunderstanding, but I'm not sure if this is accurate.  I tried using an argument of the same name in the function, something like:

```py
@cli.command
@cli.add_argument('--data', help="Some data.")
def func(data):
    """Command."""
    print(data)
```

This produces an error: `ValueError: Argument '--data declared twice.'`.  As far as I can tell, the _only_ way to pass arguments is using `*args` and `**kwargs` in the case that you use `add_argument`.

**Comment by innot** [comment]
Do I understand correctly that you would like something this:

```
def func(my_flag: RequiredFlag, my_option: RequiredOption):
    ....

```
I think adding two new Annotation keywords is the easiest and least ambiguous way to implement this as to my understanding of the argparse library the "required=true" is only applicable to flags.

There is already some code for "required" in the argparseDecorator, but I never actually exposed this, probably due to this statement from the argparse documentation:

-  **Note:** Required options are generally considered bad form because users expect options to be optional, and thus they should be avoided when possible.

But if need be I am open to adding this to argparseDecorator.

Regarding the second issue with the `ValueError: Argument '--data declared twice.'` I think this is a bug in the program as I intended it to behave like in the documentation. However there might be some issues with ambiguity there - I have to look into this.

**Comment by Chris3606** [comment]
Yes, this accurately summarizes my request :D  I saw that note in the the documentation; I will admit I disagree with the python documentation here in _certain_ cases; for example, you have 6 or 7 positional arguments I find it incredibly unintuitive to write the command as compared to required flags.

I think another decorator would work well; although I guess it should probably be compatible with the ability to set aliases in the docstring like `Option` and `Flag` are.

**Comment by innot** [comment]
I just released a new version (1.3.1) with all the additions from issues #4 and #5.
Enjoy :-)
